tests/unit/test___init__.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from commons.excepts import NoSuchModel, SelectModelRequired
+from commons.models.inference import ONNXRuntimeInferenceEngine, TorchInferenceEngine
+
+import finetuner
+
+
+@pytest.mark.parametrize(
+    'descriptor, select_model, is_onnx, expect_error',
+    [
+        ('bert-base-cased', None, False, None),
+        ('bert-base-cased', None, True, None),
+        ('openai/clip-vit-base-patch16', 'clip-text', False, None),
+        ('openai/clip-vit-base-patch16', 'clip-vision', False, None),
+        ('not a real model', None, False, NoSuchModel),
+        ('openai/clip-vit-base-patch16', None, False, SelectModelRequired),
+    ],
+)
+def test_build_model_and_embed(descriptor, select_model, is_onnx, expect_error):
+
+    if expect_error:
+        with pytest.raises(expect_error):
+            model = finetuner.build_model(
+                name=descriptor,
+                select_model=select_model,
+                is_onnx=is_onnx,
+            )
+    else:
+        model = finetuner.build_model(
+            name=descriptor, select_model=select_model, is_onnx=is_onnx
+        )
+    # encode and check the embeddings
+    model = finetuner.build_model(
+        name='bert-base-cased',
```

Comment:
is this correct to call it a second time?

Reply:
no it is not, fixed now